inc/knowbaseitem_revision.class.php

```diff
@@ -0,0 +1,348 @@
+            $where = [
+               'knowbaseitems_id' => $item->getID(),
+               'language'         => ''
+            ];
+         } else {
+            $where = [
+               'knowbaseitems_id' => $item->fields['knowbaseitems_id'],
+               'language'         => $item->fields['language']
+            ];
+         }
+
+         $nb = countElementsInTable(
+            'glpi_knowbaseitems_revisions',
+            $where
+         );
+      }
+      return self::createTabEntry(self::getTypeName($nb), $nb);
+   }
+
+   static function displayTabContentForItem(CommonGLPI $item, $tabnum=1, $withtemplate=0) {
+      self::showForItem($item, $withtemplate);
+      return true;
+   }
+
+   /**
+    * Show linked items of a knowbase item
+    *
+    * @param $item                     CommonDBTM object
+    * @param $withtemplate    integer  withtemplate param (default '')
+   **/
+   static function showForItem(CommonDBTM $item, $withtemplate='') {
+      global $DB, $CFG_GLPI;
+
+      $kbitem_id = null;
+      $item_id = $item->getID();
+      $item_type = $item::getType();
+      if (isset($_GET["start"])) {
+         $start = intval($_GET["start"]);
+      } else {
+         $start = 0;
+      }
+
+      // Total Number of revisions
+      if ($item->getType() == KnowbaseItem::getType()) {
+         $where = [
+            'knowbaseitems_id' => $item->getID(),
+            'language'         => ''
+         ];
+      } else {
+         $where = [
+            'knowbaseitems_id' => $item->fields['knowbaseitems_id'],
+            'language'         => $item->fields['language']
+         ];
+      }
+
+      $number = countElementsInTable(
+         'glpi_knowbaseitems_revisions',
+         $where
+       );
+
+      // No revisions in database
+      if ($number < 1) {
+         $no_txt = __('No revisions');
+         echo "<div class='center'>";
+         echo "<table class='tab_cadre_fixe'>";
+         echo "<tr><th>$no_txt</th></tr>";
+         echo "</table>";
+         echo "</div>";
+         return;
+      }
+      // Display the pager
+
+      Html::printAjaxPager(self::getTypeName(1), $start, $number);
+      // Output events
+      echo "<div class='center'>";
+      echo "<input type=\"button\" name=\"compare\" value=\""._sx('button', 'Compare selected revisions')."\" class=\"submit compare\">";
+      echo "<table class='tab_cadre_fixehov'>";
+      $header = '<tr>';
+      $header .= "<th title=\"" . __('Revision') . "\">#</th>";
+      $header .= "<th>&nbsp;</th>";
+      $header .= "<th>" . _('Author')  . "</th>";
+      $header .= "<th>".__('Creation date')."</th>";
+      $header .= "<th></th></tr>";
+      echo $header;
+
+      $user = new User();
+      $user->getFromDB($item->fields['users_id']);
+
+      //current contents
+      echo "<tr class='tab_bg_2'>";
+      echo "<td>(" . __('cur')  . ")</td>" .
+              "<td><input type=\"radio\" name=\"oldid\" value=\"0\" style=\"visibility:hidden\"/>" .
```

Comment:
Non, not at all. Using single ou double quotes has always been permitted, but the previous HTML specs were unclear ;)